From ccbfddb1c830b2593a22d2955c11be41251d3d4a Mon Sep 17 00:00:00 2001
From: Sandor Yu <Sandor.yu@nxp.com>
Date: Thu, 16 Nov 2017 11:48:38 +0800
Subject: [PATCH 2823/5242] MLK-16834-03: hdmi/dp phy: Enable additional PLL
 loop Amplifier.

commit  b8249c6c6ca7e6d79a0a047d18e2843b888f1954 from
https://source.codeaurora.org/external/imx/linux-imx.git

Addresses the PLL lock issue found on many devices.

Signed-off-by: Sandor Yu <Sandor.yu@nxp.com>
(cherry picked from commit 5e16126749b90e3e53fc8872b87d310ce808f84e)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/imx/hdp/API_AFE_mcu1_dp.c          |    2 +-
 .../drm/imx/hdp/API_AFE_ss28fdsoi_kiran_hdmitx.c   |    2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/imx/hdp/API_AFE_mcu1_dp.c b/drivers/gpu/drm/imx/hdp/API_AFE_mcu1_dp.c
index 799162f..e437120 100644
--- a/drivers/gpu/drm/imx/hdp/API_AFE_mcu1_dp.c
+++ b/drivers/gpu/drm/imx/hdp/API_AFE_mcu1_dp.c
@@ -320,7 +320,7 @@ void AFE_init(state_struct *state, int num_lanes, ENUM_AFE_LINK_RATE link_rate)
 	val = val | 0x0008;
 	Afe_write(state, PHY_PMA_CMN_CTRL1, val);
 
-	Afe_write(state, CMN_DIAG_PLL0_TEST_MODE, 0x0020);
+	Afe_write(state, CMN_DIAG_PLL0_TEST_MODE, 0x0022);
 	Afe_write(state, CMN_PSM_CLK_CTRL, 0x0016);
 
 	phy_cfg_24mhz(state, num_lanes);
diff --git a/drivers/gpu/drm/imx/hdp/API_AFE_ss28fdsoi_kiran_hdmitx.c b/drivers/gpu/drm/imx/hdp/API_AFE_ss28fdsoi_kiran_hdmitx.c
index 861cc63..be58b46 100644
--- a/drivers/gpu/drm/imx/hdp/API_AFE_ss28fdsoi_kiran_hdmitx.c
+++ b/drivers/gpu/drm/imx/hdp/API_AFE_ss28fdsoi_kiran_hdmitx.c
@@ -205,7 +205,7 @@ int phy_cfg_hdp_ss28fdsoi(state_struct *state, int num_lanes, VIC_MODES vicMode,
 			Afe_write(state, 0x40E8 | (i << 9), 0x007F);
 	}
 	/* register CMN_DIAG_PLL0_TEST_MODE */
-	Afe_write(state, 0x01C4, 0x0020);
+	Afe_write(state, 0x01C4, 0x0022);
 	/* register CMN_PSM_CLK_CTRL */
 	Afe_write(state, 0x0061, 0x0016);
 
-- 
1.7.9.5

